src/pipelines/worldpop/worldpop_population.py

```diff
@@ -0,0 +1,101 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import zipfile
+from io import BytesIO
+from functools import partial
+from typing import Any, Dict, List, Tuple
+
+import requests
+from tqdm import tqdm
+from pandas import DataFrame, Series, read_csv, isnull
+
+from lib.concurrent import thread_map
+from lib.net import download
+from lib.pipeline import DataSource
+from lib.cast import age_group
+
+
+class WorldPopPopulationDataSource(DataSource):
```

Comment:
Where is the output of this pipeline?  I don't see demographics.csv in this PR.

Reply:
It would be great if this could be unit tested.  If it's too difficult, at least point me to snippets of input and output to help me verify.  I can think about setting up a unittest framework to try to make writing a unit test as easy as adding snippets of input and output (anyone who adds a data source should be examining the input and output anyway, so hopefully it can be very low cost).

Reply:
demographics.csv is there, it's just very, very large (10K line change!). That's because I'm changing the schema, so every line is affected.

I totally agree about unit testing. Or at the very least some kind of "summary statement" to describe what has changed about the data. Let's think about this holistically and fix it for all data sources, rather than one at a time though.

Reply:
I'll think about a general framework to make this easier for all data sources, but would it probably still involve providing some input/data data for each data source.  Maybe the ingestion of this test data per data source can be (semi)automated, but whenever someone changes the expected output for a data source they should still look at and verify the test data.